### DISCUSSION TITLE
can't handle key event after update to v11?

### DISCUSSION THREAD

**Discussion by uno-studio** [AvaloniaUI/Avalonia#12550] [Q&A] [answered]
after updating to v11 my user control can't capture key board events. although at 0.10 everything works fine.
OS: Windows 10
here is the code:

`protected async override void OnKeyDown(KeyEventArgs e){ }`
`protected async override void OnKeyUp(KeyEventArgs e){ }`

OnTextInput event also not working in V11

**Comment by maxkatz6** [comment]
Check devtools to see where these events are raised and where they go.

**Reply by uno-studio** [reply]
after checking, for mouse events, works properly:
![1](https://github.com/AvaloniaUI/Avalonia/assets/43975116/4f2b479c-b58a-4742-9414-a0eebd18d6aa)

This is the key event, redirected to another listbox.
![2](https://github.com/AvaloniaUI/Avalonia/assets/43975116/6a528fe2-fa77-4102-8c21-23bb27d2bd56)

**Reply by timunie** [reply]
does the control has KeyboardFocus?

**Reply by uno-studio** [reply]
there's a problem. when i click on usecontrol, focus is always listbox
![2023-08-16 14 58 36](https://github.com/AvaloniaUI/Avalonia/assets/43975116/fe7bf1d7-ed00-4bc5-aa17-bee632952887)

**Reply by timunie** [reply, marked as answer]
Set Focusable=True

**Reply by uno-studio** [reply]
thanks, working
